worker/src/ocall_bridge/attestation_ocall_impl.rs

```diff
@@ -0,0 +1,163 @@
+
+        let (p_sig_rl, sig_rl_size) = vec_to_c_pointer_with_len(revocation_list);
+
+        let ret =
+            unsafe { sgx_calc_quote_size(p_sig_rl, sig_rl_size, &mut real_quote_len as *mut u32) };
+
+        if ret != sgx_status_t::SGX_SUCCESS {
+            error!("   sgx_calc_quote_size failed. {}", ret);
+            return (ret, sgx_report_t::default(), Vec::new());
+        }
+
+        debug!("    Quote size = {}", real_quote_len);
+
+        let p_report = &report as *const sgx_report_t;
+        let p_spid = &spid as *const sgx_spid_t;
+        let p_nonce = &quote_nonce as *const sgx_quote_nonce_t;
+
+        let mut qe_report = sgx_report_t::default();
+        let p_qe_report = &mut qe_report as *mut sgx_report_t;
+
+        const RET_QUOTE_BUF_LEN: usize = 2048;
+        let mut return_quote_buf = [0u8; RET_QUOTE_BUF_LEN];
+        let p_quote = return_quote_buf.as_mut_ptr();
+
+        if real_quote_len > RET_QUOTE_BUF_LEN as u32 {
+            error!(
+                "   effective quote length ({}) exceeds buffer size ({})",
+                real_quote_len, RET_QUOTE_BUF_LEN
+            );
+            return (
+                sgx_status_t::SGX_ERROR_FAAS_BUFFER_TOO_SHORT,
+                sgx_report_t::default(),
+                Vec::new(),
+            );
+        }
+
+        let ret = unsafe {
+            sgx_get_quote(
+                p_report,
+                quote_type,
+                p_spid,
+                p_nonce,
+                p_sig_rl,
+                sig_rl_size,
+                p_qe_report,
+                p_quote as *mut sgx_quote_t,
+                real_quote_len,
+            )
+        };
+
+        if ret != sgx_status_t::SGX_SUCCESS {
+            error!("    sgx_get_quote failed. {}", ret);
+            return (ret, sgx_report_t::default(), Vec::new());
+        }
+
+        (
+            sgx_status_t::SGX_SUCCESS,
+            qe_report,
+            Vec::from(&return_quote_buf[..real_quote_len as usize]),
+        )
+    }
+
+    fn get_update_info(
+        &self,
+        platform_blob: sgx_platform_info_t,
+        enclave_trusted: i32,
+    ) -> (sgx_status_t, sgx_update_info_bit_t) {
+        let mut update_info: sgx_update_info_bit_t = sgx_update_info_bit_t::default();
+
+        let result = unsafe {
+            sgx_report_attestation_status(
+                &platform_blob as *const sgx_platform_info_t,
+                enclave_trusted,
+                &mut update_info as *mut sgx_update_info_bit_t,
+            )
+        };
+
+        (result, update_info)
+    }
+}
+
+fn lookup_ipv4(host: &str, port: u16) -> SocketAddr {
+    use std::net::ToSocketAddrs;
+
+    let addrs = (host, port).to_socket_addrs().unwrap();
+    for addr in addrs {
+        if let SocketAddr::V4(_) = addr {
+            return addr;
+        }
+    }
+
+    unreachable!("Cannot lookup address");
+}
+
+fn vec_to_c_pointer_with_len<A>(input: Vec<A>) -> (*const A, u32) {
```

Comment:
Yes, this was more meant as inspiration. It might be an overkill.

Well, any crate that is strictly no_std would run in both envs. But we don't have any crates std-dependant code that runs in both envs.